ducktape/cluster/remoteaccount.py

```diff
@@ -142,97 +317,182 @@ def kill_process(self, process_grep_str, clean_shutdown=True, allow_fail=False):
-        and the CalledProcessError is re-raised. If allow_fail is True, the
-        CalledProcess error will not be re-raised and the exit status of the
-        process will be returned instead.
+    def isdir(self, path):
+        try:
+            # stat should follow symlinks
+            path_stat = self.sftp_client.stat(path)
+            return stat.S_ISDIR(path_stat.st_mode)
+        except:
+            # TODO figure out which errors are legit (e.g. if file does not exist, what will sftp.stat do?)
+            is_file = False
+
+        return is_file
+
+    def exists(self, path):
+        """Test that the path exists, but don't follow symlinks."""
+        try:
+            # stat follows symlinks and tries to stat the actual file
+            self.sftp_client.lstat(path)
+            return True
+        except IOError:
+            return False
+
+    def isfile(self, path):
+        """Imitates semantics of os.path.isfile
+
+        :path Path to the thing to check
+        :return True iff path is a file or a symlink to a file, else False. Note False can mean path does not exist.
         """
         try:
-            if self.logger is not None:
-                self.logger.debug("Trying to run remote command: " + cmd)
-            subprocess.check_output(cmd, shell=True, stderr=subprocess.STDOUT)
-            return 0
-        except subprocess.CalledProcessError as e:
-            # Depending on the user, failure of remote commands may be part of normal usage patterns (e.g. if used in
-            # a "wait_until" loop). So, log it, but don't make it too scary.
-            if self.logger is not None:
-                self.logger.debug("Error running remote command: " + cmd)
-                self.logger.debug(e.output)
-
-            if allow_fail:
-                return e.returncode
-            raise e
+            # stat should follow symlinks
+            path_stat = self.sftp_client.stat(path)
+            return stat.S_ISREG(path_stat.st_mode)
+        except:
+            # TODO figure out which errors are legit (e.g. if file does not exist, what will sftp.stat do?)
+            is_file = False
+
+        return is_file
+
+    def open(self, path, mode='r'):
+        return self.sftp_client.open(path, mode)
+
+    def create_file(self, path, contents):
+        """Create file at path, with the given contents.
+
+        If the path already exists, it will be overwritten.
+        """
+        # TODO: what should semantics be if path exists? what actually happens if it already exists?
+        # TODO: what happens if the base part of the path does not exist?
+        with self.sftp_client.open(path, "w") as f:
+            f.write(contents)
+
+    def mkdir(self, path, mode=0755):
+
+        self.sftp_client.mkdir(path, mode)
+
+    def mkdirs(self, path, mode=0755):
+        self.ssh("mkdir -p %s && chmod %o %s" % (path, mode, path))
+
+    def remove(self, path, allow_fail=False):
```

Comment:
Hmm, I know we use `allow_fail` elsewhere (something I'm not ecstatic about since it's kinda wordy), but with this a `force` would be way more intuitive since I'm so used to the `-f` on `rm`.